pkg/storage/storage.go

```diff
@@ -56,3 +57,29 @@ func SetStorage(storageEngine Storage) error {
 	storage = storageEngine
 	return nil
 }
+
+// GetStorage returns the primary storage for events.
+func GetStorage() (Storage, error) {
+	if storage == nil {
+		return nil, fmt.Errorf("no storage engine assigned")
+	}
+	return storage, nil
+}
+
+// SetStorage sets the desired storage engine for read-only events. Switching to a new
+// storage engine implies garbage collecting the old one, with possible loss of
+// pending events if not flushed correctly
+func SetStorageAsync(storageEngine Storage) error {
```

Comment:
Maybe name this `SetAsyncStorage`, also copy-paste in first word of the doc comment :)